docs/template_plugin/tests/functional/op_reference/experimental_detectron_roi_feature_extractor.cpp

```diff
@@ -0,0 +1,135 @@
+using ExperimentalROI = op::v6::ExperimentalDetectronROIFeatureExtractor;
+
+struct ExperimentalROIFunctional {
+    std::shared_ptr<Function> create_function(const std::vector<Tensor>& ed_inputs,
+                                              const std::vector<Tensor>& results) {
+        Attrs attrs;
+        attrs.aligned = false;
+        attrs.output_size = 3;
+        attrs.sampling_ratio = 2;
+        attrs.pyramid_scales = {4};
+
+        auto input = std::make_shared<op::Parameter>(element::f32, Shape{2, 4});
+        auto pyramid_layer0 = std::make_shared<op::Parameter>(element::f32, Shape{1, 2, 2, 3});
+
+        auto roi = std::make_shared<ExperimentalROI>(NodeVector{input, pyramid_layer0}, attrs);
+
+        auto fun = std::make_shared<Function>(OutputVector{roi->output(0), roi->output(1)},
+                                              ParameterVector{input, pyramid_layer0});
+        return fun;
+    }
+};
+
+struct ExperimentalROIParams {
+    ExperimentalROIParams(const std::shared_ptr<ExperimentalROIFunctional>& functional,
+                          const std::vector<Tensor>& experimental_detectron_roi_feature_inputs,
+                          const std::vector<Tensor>& expected_results,
+                          const std::string& test_case_name)
+        : function{functional},
+          inputs{experimental_detectron_roi_feature_inputs},
+          expected_results{expected_results},
+          test_case_name{test_case_name} {}
+
+    std::shared_ptr<ExperimentalROIFunctional> function;
+    std::vector<Tensor> inputs;
+    std::vector<Tensor> expected_results;
+    std::string test_case_name;
+};
+
+class ReferenceExperimentalROILayerTest : public testing::TestWithParam<ExperimentalROIParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = params.function->create_function(params.inputs, params.expected_results);
+        inputData.reserve(params.inputs.size());
+        refOutData.reserve(params.expected_results.size());
+        for (auto& input_tensor : params.inputs) {
+            inputData.push_back(input_tensor.data);
+        }
+        for (auto& expected_tensor : params.expected_results) {
+            refOutData.push_back(expected_tensor.data);
+        }
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<ExperimentalROIParams>& obj) {
+        auto param = obj.param;
+        return param.test_case_name;
+    }
+};
+
+TEST_P(ReferenceExperimentalROILayerTest, ExperimentalROIWithHardcodedRefs) {
+    Exec();
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_ExperimentalROI_With_Hardcoded_Refs,
+    ReferenceExperimentalROILayerTest,
+    ::testing::Values(
+        ExperimentalROIParams(
+            std::make_shared<ExperimentalROIFunctional>(),
+            std::vector<Tensor>{Tensor(Shape{2, 4},
+                                       ngraph::element::f32,
```

Comment:
As I see `ExperimentalDetectronROIFeatureExtractor` supports not only f32.
Please add tests for all possible data types.

Reply:
Done.